bindings/py/nupic/bindings/regions/PyRegion.py

```diff
@@ -0,0 +1,359 @@
+  @abstractmethod
+  def __init__(self, *args, **kwars):
+    """Initialize the node with creation parameters from the node spec
+
+    Should be implemented by subclasses (unless there are no creation params)
+    """
+
+
+  @abstractmethod
+  def initialize(self, inputs, outputs):
+    """Initialize the node after the network is fully linked
+
+    It is called once by NuPIC before the first call to compute(). It is
+    a good place to perform one time initialization that depend on the inputs
+    and/or outputs. The region may also remember its inputs and outputs here
+    because they will not change.
+
+    inputs: dict of numpy arrays (one per input)
+    outputs: dict of numpy arrays (one per output)
+    """
+
+
+  @abstractmethod
+  def compute(self, inputs, outputs):
+    """Perform the main computation
+
+    This method is called in each iteration for each phase the node supports.
+
+    inputs: dict of numpy arrays (one per input)
+    outputs: dict of numpy arrays (one per output)
+    """
+
+
+  def getOutputElementCount(self, name):
+    """Return the number of elements in the output of a single node
+
+    If the region has multiple nodes (all must have the same output
+    size) then just the number of output elements of a single node
+    should be returned.
+
+    name: the name of the output
+    """
+    raise NotImplementedError()
+
+
+  def getParameter(self, name, index):
+    """Default implementation that return an attribute with the requested name
+
+    This method provides a default implementation of getParameter() that simply
+    returns an attribute with the parameter name. If the Region conceptually
+    contains multiple nodes with separate state the 'index' argument is used
+    to request a parameter of a specific node inside the region. In case of
+    a region-level parameter the index should be -1
+
+    The implementation prevents accessing parameters names that start with '_'.
+    It may be better to enforce this convention at the node spec level.
+
+    name: name of requested parameter
+    index: index of node inside the region (if relevant)
+
+    """
+    if name.startswith('_'):
+      raise Exception('Parameter name must not start with an underscore')
+
+    value = getattr(self, name)
+    return value
+
+
+  def getParameterArrayCount(self, name, index):
+    """Default implementation that return the length of the attribute
+
+    This default implementation goes hand in hand with getParameterArray().
+    If you override one of them in your subclass, you should probably override
+    both of them.
+
+    The implementation prevents accessing parameters names that start with '_'.
+    It may be better to enforce this convention at the node spec level.
+
+    name: name of requested parameter
+    index: index of node inside the region (if relevant)
+    """
+    if name.startswith('_'):
+      raise Exception('Parameter name must not start with an underscore')
+
+
+    v = getattr(self, name)
```

Comment:
`v` not used.  I'm assuming this is done as sort of an integrity check.  In which case, an `assert name in self` with a descriptive assertion message would be better.